### PR TITLE
[Merged by Bors] - feat port: Algebra.Order.Group.InjSurj

### PR DESCRIPTION
**Pull request by ChrisHughes24**
655994e298904d7e5bbd1e18c95defd7b543eb94

No problems here

### CODE REVIEW

File: Mathlib/Algebra/Order/Group/InjSurj.lean [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+
+variable {α β : Type _}
+
+/-- Pullback an `ordered_comm_group` under an injective map.
+See note [reducible non-instances]. -/
+@[reducible,
+  to_additive
+      "Pullback an `ordered_add_comm_group` under an injective map."]
+def Function.Injective.orderedCommGroup [OrderedCommGroup α] {β : Type _} [One β] [Mul β] [Inv β]
```
Comment (ChrisHughes24) [inline Mathlib/Algebra/Order/Group/InjSurj.lean:22]:
```suggestion
      "Pullback an `OrderedAddCommGroup` under an injective map."]
def Function.Injective.orderedCommGroup [OrderedCommGroup α] {β : Type _} [One β] [Mul β] [Inv β]
```

File: Mathlib/Algebra/Order/Group/InjSurj.lean [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import Mathlib.Algebra.Order.Group.Instances
+
+/-!
+# Pull back ordered groups along injective maps.
+-/
+
+
+variable {α β : Type _}
+
+/-- Pullback an `ordered_comm_group` under an injective map.
```
Comment (ChrisHughes24) [inline Mathlib/Algebra/Order/Group/InjSurj.lean:17]:
```suggestion
/-- Pullback an `OrderedCommGroup` under an injective map.
```

File: Mathlib/Algebra/Order/Group/InjSurj.lean [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+def Function.Injective.orderedCommGroup [OrderedCommGroup α] {β : Type _} [One β] [Mul β] [Inv β]
+    [Div β] [Pow β ℕ] [Pow β ℤ] (f : β → α) (hf : Function.Injective f) (one : f 1 = 1)
+    (mul : ∀ x y, f (x * y) = f x * f y) (inv : ∀ x, f x⁻¹ = (f x)⁻¹)
+    (div : ∀ x y, f (x / y) = f x / f y) (npow : ∀ (x) (n : ℕ), f (x ^ n) = f x ^ n)
+    (zpow : ∀ (x) (n : ℤ), f (x ^ n) = f x ^ n) : OrderedCommGroup β :=
+  { PartialOrder.lift f hf, hf.orderedCommMonoid f one mul npow,
+    hf.commGroup f one mul inv div npow zpow with }
+#align function.injective.ordered_comm_group Function.Injective.orderedCommGroup
+
+/-- Pullback a `linear_ordered_comm_group` under an injective map.
```
Comment (ChrisHughes24) [inline Mathlib/Algebra/Order/Group/InjSurj.lean:31]:
```suggestion
/-- Pullback a `LinearOrderedCommGroup` under an injective map.
```

**Review by Ruben-VandeVelde — APPROVED** [review]

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**